integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/IntegrasjonspunktImpl.java

```diff
@@ -83,8 +85,7 @@ public GetCanReceiveMessageResponseType getCanReceiveMessage(@WebParam(name = "G
         boolean isDpv = false;
         if (certificateAvailable) {
             Audit.info("CanReceive = true", marker);
-        } else if (hasMshEndpoint()) {
-            mshCanReceive = mshClient.canRecieveMessage(organisasjonsnummer);
+        } else if (hasMshEndpoint() && (mshCanReceive = mshClient.canRecieveMessage(organisasjonsnummer))) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract the assignment out of this expression. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)